ada/ast.py

```diff
@@ -15798,6 +15801,21 @@ def put_image_equation():
                  eq_prop=BaseTypeDecl.matching_formal_type)
         )
 
+    @langkit_property(return_type=Equation, dynamic_vars=[env, origin])
+    def index_equation():
+        """
+        Return the xref equation for the `Index` attribute.
+        """
+        typ = Var(
```

Comment:
Unfortunately this won't handle the following case:
```
entry E (Entry_Id_Mod);
pragma Precondition (E'Index < 2);
```
You should instead use an approach based on lexical environments.